Telerik.JustMock/Core/MocksRepository.cs

```diff
@@ -415,15 +416,18 @@ internal void DispatchInvocation(Invocation invocation)
 
 			if (!methodMockProcessed)
 			{
+				// We have to be cearfull for the potential exception throwing in the assertion context, so skip CallOriginalBehavior processing
 				var mock = invocation.MockMixin;
 				if (mock != null)
 				{
-					foreach (var behavior in mock.FallbackBehaviors)
-						behavior.Process(invocation);
+                    mock.FallbackBehaviors
+                        .Where(behavior => !invocation.InAssertSet || !(behavior is CallOriginalBehavior))
+                        .ToList()
+                        .ForEach(behavior => behavior.Process(invocation));
```

Comment:
As we talked a few days earlier, LINQ queries are hard to debug. To improve the debug experience a variable can be introduced with the filtered behaviors for processing.

Reply:
@mihail-vladov, fixed with [fddf417](https://github.com/telerik/JustMockLite/pull/39/commits/fddf417f417b749148a45ea863e9b37874674e50)